### PR TITLE
Feature/hw49

### PR DESCRIPTION
**Pull request by fedorchaklm**
1. Перевести проект на TypeScript
2. Розбити проект на layers: controllers, routes, services, repository  
3.  Додати env файли. Support development and production environments. В package.json мають бути додані відповідні скріпти для запуску

Env variables: PORT, LOG_FILE

### CODE REVIEW

File: src/controllers/cart.controller.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { RequestWithUserId } from "src/types/types";
+import * as cartService from "../services/cart.service";
+import { Request, Response } from "express";
+
+export const addProductToCart = (req: Request, res: Response) => {
```
Comment (VitalyaIlin) [inline src/controllers/cart.controller.ts:5]: can you set type `RequestWithUserId` for the `req` then this type definition won't be needed below.
`req: RequestWithUserId`

File: src/repositories/user.repository.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { users } from "../storage";
+import { User } from "src/types/types";
+
+export const addUser = (user: User) => {
```
Comment (VitalyaIlin) [inline src/repositories/user.repository.ts]: add the type that the function returns

File: src/services/product.service.ts [hunk 3]
```diff
@@ -1,55 +1,50 @@
...
+import { Product, ProductCsv } from "src/types/types";
 
 const __filename = fileURLToPath(import.meta.url);
 const __dirname = path.dirname(__filename);
 const productsStoreFilePath = path.join(__dirname, "..", "products.store.json");
 
-router.get("/", (_, res) => {
-  res.status(200).json(products);
-});
+export const getAllProducts = () => {
```
Comment (VitalyaIlin) [inline src/services/product.service.ts]: add the type that the function returns

**Review by VitalyaIlin — COMMENTED** [review]
сheck commens please

File: src/repositories/user.repository.ts (hunk 2 above)
Comment (fedorchaklm) [reply src/repositories/user.repository.ts:4]: done

File: src/controllers/cart.controller.ts (hunk 1 above)
Comment (fedorchaklm) [reply src/controllers/cart.controller.ts:5]: image
I'm having an error when using type `RequestWithUserId` for the `req`.
I decided to create a helper function 'getUserId' and remove this 'RequestWithUserId' type.

File: src/repositories/user.repository.ts (hunk 2 above)
Comment (fedorchaklm) [reply src/repositories/user.repository.ts]: done

File: src/services/product.service.ts (hunk 3 above)
Comment (fedorchaklm) [reply src/services/product.service.ts]: done

File: .env.examples [hunk 4]
```diff
@@ -0,0 +1,2 @@
+PORT
+LOG_FILE
```
Comment (VitalyaIlin) [inline .env.examples:2]: Typically, the file format as follows:
PORT=
LOG_FILE=

File: .env.development [hunk 5]
```diff
@@ -0,0 +1,2 @@
+PORT=3000
```
Comment (VitalyaIlin) [inline .env.development]: remove this file from the pull request

File: .env.production [hunk 6]
```diff
@@ -0,0 +1,2 @@
+PORT=80
```
Comment (VitalyaIlin) [inline .env.production]: remove this file from the pull request

File: package.json [hunk 7]
```diff
@@ -2,10 +2,11 @@
   "name": "nodejs",
   "version": "1.0.0",
   "description": "",
-  "main": "index.js",
+  "main": "src/index.ts",
   "type": "module",
   "scripts": {
-    "dev": "nodemon index.js",
+    "dev": "cross-env NODE_ENV=development ts-node --esm src/index.ts",
+    "start": "cross-env NODE_ENV=production ts-node --esm src/index.ts",
```
Comment (VitalyaIlin) [inline package.json:9]: In production, the generated .js files are being run

**Review by VitalyaIlin — APPROVED** [review]

**Pull request merged**